File: src/main/java/com/example/lat/model/AttendanceRecord.java
package com.example.lat.model;

import java.time.LocalDate;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.Data;

@Data
@Entity
public class AttendanceRecord {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Student student;

    private LocalDate date;

    @Enumerated(EnumType.STRING)
    private Status status;

    public enum Status {
        PRESENT, ABSENT
    }
}

File: MyApplication/app/src/main/java/com/example/attendanceapp/model/AttendanceRecord.kt
package com.example.attendanceapp.model

import java.time.LocalDate

data class AttendanceRecord(
    val id: Long? = null,
    val student: Student,
    val date: String, // We'll send as string to match backend LocalDate
    val status: AttendanceStatus
)

enum class AttendanceStatus {
    PRESENT, ABSENT
}

// For sending attendance data to backend
data class AttendanceSubmission(
    val date: String,
    val records: List<AttendanceRecord>
)
